src/target/riscv/riscv.c

```diff
@@ -5365,14 +5318,39 @@ static struct reg_arch_type riscv_reg_arch_type = {
 	.set = register_set
 };
 
-struct csr_info {
-	unsigned number;
-	const char *name;
-};
-
-static int cmp_csr_info(const void *p1, const void *p2)
+static int initialize_custom_register_names(struct list_head *expose_custom,
+		struct reg_name_table *custom_register_names)
 {
-	return (int) (((struct csr_info *)p1)->number) - (int) (((struct csr_info *)p2)->number);
+	unsigned int custom_regs_num = 0;
+	if (!list_empty(expose_custom)) {
+		range_list_t *entry;
+		list_for_each_entry(entry, expose_custom, list)
+			custom_regs_num += entry->high - entry->low + 1;
+	}
+
+	if (!custom_regs_num)
+		return ERROR_OK;
+
+	custom_register_names->reg_names = calloc(custom_regs_num, sizeof(struct reg_name));
+	custom_register_names->names_num = custom_regs_num;
```

Comment:
Move this after the reg_names check, so if the error happens we don't falsely lately believe names_num>0

Reply:
Addressed.